core/app/models/spree/order.rb

```diff
@@ -561,7 +557,7 @@ def refresh_shipment_rates
     end
 
     def shipping_eq_billing_address?
-      (bill_address.empty? && ship_address.empty?) || bill_address.same_as?(ship_address)
+      (bill_address.empty? && ship_address.empty?) || bill_address == ship_address
```

Comment:
since we are doing equality, if they are both empty, won't that be caught by the new equality check?